src/altinn-app-frontend/src/features/form/containers/RepeatingGroupTable.tsx

```diff
@@ -540,21 +552,21 @@ export function RepeatingGroupTable({
                 ].some((component: ILayoutComponent | ILayoutGroup) => {
                   return childElementHasErrors(component, index);
                 });
-                const items: IMobileTableItem[] = [];
-                components.forEach((component) => {
-                  const childId =
-                    (component as any).baseComponentId || component.id;
-                  if (tableHeaderComponents.includes(childId)) {
-                    items.push({
+                const items: IMobileTableItem[] = tableRenderComponents.map(
+                  (component: ILayoutComponent) => {
+                    let label = '';
+                    if (component?.textResourceBindings?.tableTitle) {
+                      label = component.textResourceBindings.tableTitle;
+                    } else if (component?.textResourceBindings?.title) {
```

Comment:
Isn't this functionality duplicated now? :thinking: I haven't looked thoroughly into it, but it might be better to spend some time deduplicating this functionality. Boy scout rule: Always leave the code better than you found it. If the mobile table and the desktop table differs wildly, it might be time to introduce something generic that can render both a mobile and a desktop table.

Reply:
It was already duplicated even more before, but I can possibly fix this too